src/plugins/risk_measures.jl

```diff
@@ -43,3 +43,231 @@ function adjust_probability(measure::WorstCase,
+    future cost pᵀ̃z. If we maximize reward, we would have ̃p ∈ argmin{̃pᵀ̃z}.
+2. ̄a is the uniform distribution
+3. r is a user specified radius - the larger the radius, the more conservative
+   the policy.
+
+Note: the largest radius that will work with S scenarios is sqrt((S-1)/S).
+=#
+
+"""
+    DRO(radius::Float64)
+
+The distributionally robust SDDP risk measure of
+
+Philpott, A., de Matos, V., Kapelevich, L. Distributionally robust SDDP.
+Computational Management Science (2018) 165:431-454.
+"""
+struct DRO <: AbstractRiskMeasure
+    radius::Float64
+end
+
+function adjust_probability(measure::DRO,
+                            risk_adjusted_probability::Vector{Float64},
+                            original_probability::Vector{Float64},
+                            noise_support::Vector,
+                            objective_realizations::Vector{Float64},
+                            is_minimization::Bool)
+    m = length(objective_realizations)
+    if !(original_probability ≈ fill(1.0 / m, m))
+        error("Current implementation of DRO assumes a uniform nominal distriution.")
+    end
+    if abs(measure.radius) < 1e-9 ||
+            Statistics.std(objective_realizations, corrected=false) < 1e-9
+        # Don't do any DRO reweighting if the radius is small or the variance
+        # is too low. Use Expectation instead.
+        return adjust_probability(
+           Expectation(), risk_adjusted_probability, original_probability,
+           noise_support, objective_realizations, is_minimization)
+   end
+    # Sort future costs/rewards
+    if is_minimization
+        perm = sortperm(objective_realizations)
+        z = -objective_realizations[perm]
+    else
+        perm = sortperm(objective_realizations, rev = true)
+        z = objective_realizations[perm]
+    end
+    # Take a permuted view of `risk_adjusted_probability` so we can refer to
+    # `p[i]` instead of `risk_adjusted_probability[perm[i]]`.
+    p = view(risk_adjusted_probability, perm)
+    # Compute the new probabilities according to Algorithm (2) of the Philpott
+    # et al. paper.
+    # Step (1):
+    @inbounds for k in 0:m-2
+        # Step (1a):
+        z_bar = sum(z[i] for i in (k+1):m) / (m - k)
+        s = sqrt(sum(z[i]^2 - z_bar^2 for i in (k+1):m) / (m - k))
+        # Step (1b): note that we cache a couple of terms that don't depend on i
+        #            to speed things up.
+        term_1 = 1 / (m - k)
+        term_2 = sqrt((m - k) * measure.radius^2 - k / m) / ((m - k) * s)
+        # We really should set p[i] = 0 for i = 1, ..., k. But since we don't
+        # touch p[k-1] again, we can just set the k'th element to 0.
+        if k > 0
+            p[k] = 0.0
+        end
+        @inbounds for i in (k+1):m
+            # TODO(lkapelevich): in the paper, it suggests that it should be
+            # p[i] = term_1 + term_2 * (z[i] - z_bar)
+            p[i] = term_1 + term_2 * (z_bar - z[i])
```

Comment:
@lkapelevich, I'm re-implementing DRO and tidying some things up. For my own pedagogical interest, I implemented the paper verbatim rather than looking at your implementation. What's the story with this difference in sign?

Reply:
Will take a look at this later!

Reply:
Don't worry. I figured it out.